### PR TITLE
Fix sharing of reactive power with for multiple generators with ReactivePowerControl

### PR DESCRIPTION
**Pull request by caioluke**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

Reactive power is always equally split between generators on a bus controlling the reactive power on a branch. Thus, ignoring the reactive keys. 

**What is the new behavior (if this is a feature change)?**
The reactive keys will not only work for splitting the reactive power between different buses but also for different generators on the same bus.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java [hunk 1]
```diff
@@ -538,6 +586,11 @@ protected static double dispatchQ(List<LfGenerator> generatorsWithControl, boole
             case K_EQUAL_PROPORTION -> allGeneratorsHavePlausibleReactiveLimits(generatorsWithControl)
                     ? splitDispatchQWithEqualProportionOfK(generatorsWithControl, qToDispatch)
                     : splitDispatchQ(generatorsWithControl, qToDispatch); // fallback to q dispatch
+            case REACTIVE_KEYS -> allGeneratorsHaveReactiveKeys(generatorsWithControl)
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java:589]: I am not convinced by your fix. Indeed, during calculation, if reactive keys are present, we take them (`createReactiveKeys`). If not, we have two fall backs. `K_EQUAL_PROPORTION` is a bit special and must not be changed, but I think that the good solution is to adapt the `Q_EQUAL_PROPORTION` that seems to be too simple compared to what we do for calculation. To be confirmed by @geofjamg, but I think that you have to follow the same logic as the logic we have for creation.

Comment (caioluke) [reply src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java:589]: I agree with you, there is definetely something to discuss about my fix.
The thing is that reactive keys for calculations created by `createReactiveKeys` are used to split reactive power between **buses**. Whereas here `reactivePowerDispatchMode` controls how reactive power is split within a bus, between **generators**. I don't know if there are cases which we would like to split reactive power between buses in a certain way and in another way between generators...

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java:589]: Yes of course, but as you said, we must be coherent...

**Review by annetill — APPROVED** [review]
Thanks a lot!

**Pull request merged**